core/src/validator.rs

```diff
@@ -623,6 +629,29 @@ impl Validator {
 
         let start_time = Instant::now();
 
+        if !ledger_path.is_dir() {
+            return Err(anyhow!(
+                "ledger directory does not exist or is not accessible: {ledger_path:?}"
+            ));
+        }
+        let genesis_config = load_genesis(config, ledger_path)?;
+        metrics_config_sanity_check(genesis_config.cluster_type)?;
+
+        // Measure the PoH hash rate as early as possible to minimize
+        // contention with other threads
+        //
+        // Skip the check for Development clusters as this will be the type for
+        // tests or for clusters that we don't know what the rate will be
+        // without a rebuilt Bank
+        let my_poh_hashes_per_second =
+            match (genesis_config.cluster_type, !config.no_poh_speed_test) {
```

Comment:
matching !no with false arm breaks my brain

Reply:
Ha, I initially did the following but cleared it
```rust
let do_speed_test = !config.no_speed_test
let my_poh_hashes_per_second = match ...
```
Naming a separate variable makes the not more visible so can switch it back if that's agreeable

Reply:
ha, I'm cool with whatever